packages/marshal/src/rankOrder.js

```diff
@@ -0,0 +1,300 @@
+// @ts-check
+
+import { assert, details as X, q } from '@agoric/assert';
+import { getTag } from './helpers/passStyle-helpers.js';
+import { nameForPassableSymbol } from './helpers/symbol.js';
+import { Far } from './make-far.js';
+import { makeTagged } from './makeTagged.js';
+import { passStyleOf } from './passStyleOf.js';
+import { sameValueZero } from './pureCopy.js';
+
+const {
+  getOwnPropertyDescriptors,
+  fromEntries,
+  entries,
+  setPrototypeOf,
+} = Object;
+
+const { ownKeys } = Reflect;
+
+const firstSymbol = Symbol.for('');
+const firstTagged = makeTagged('', null);
+const aFar = Far('after taggeds', {});
+const anError = new Error('error sigil');
+const aPromise = Promise.resolve('promise sigil');
+
+/**
+ * Aside from 'undefined', lists the PassStyles from `types.js` in the same
+ * order as they are declared there. This list and that declaration must be
+ * co-maintained.
```

Comment:
> co-maintained.

gives me the willies.  Why not use a key somewhere so that at least they aren't both order-sensitive, or at least so that the key-to-order mapping is only defined once?

Reply:
The other side is the type declaration at https://github.com/Agoric/agoric-sdk/blob/1c220c4454a9ae5594ab145e27b6ceecc6c3f6cb/packages/marshal/src/types.js#L7-L26 . Given that, does this suggestion still make sense?

Reply:
> does this suggestion still make sense?

No, as you've explained it, no need to change.  But I think the "co-maintained" warning is more scary than it needs to be.  Maybe something like: "the PassStyle declaration (cf. somewhere) should be kept in sync with this data structure that implements it".  YMMV, and I won't be upset if you make no change.

Reply:
Done.